community/gill-jito-airdrop/anchor/scripts/extract-wallet.ts

```diff
@@ -0,0 +1,193 @@
+    const allWallets = deployWallet ? [deployWallet, ...testWallets] : testWallets;
+    
+    const wallet = allWallets.find(w => w.name === walletName);
+    if (!wallet) {
+      console.error(`❌ Wallet '${walletName}' not found`);
+      return false;
+    }
+    
+    const output = outputPath || wallet.keypairFile;
+    const keypairArray = wallet.secretKey.array;
+    fs.writeFileSync(output, JSON.stringify(keypairArray, null, 2));
+    
+    console.log(`✅ Extracted ${walletName} to ${output}`);
+    return true;
+  } catch (error) {
+    console.error(`❌ Error extracting wallet '${walletName}':`, error);
+    return false;
+  }
+}
+
+function loadWallets(): { deployWallet: any | null, testWallets: any[] } {
+  try {
+    if (!fs.existsSync("test-wallets.json")) {
+      return { deployWallet: null, testWallets: [] };
+    }
+    
+    const data = fs.readFileSync("test-wallets.json", 'utf8');
+    const walletsData: TestWalletsData = JSON.parse(data);
+    
+    const deployWallet = walletsData.wallets.find(w => w.isDeployWallet || w.name === "deploy-wallet");
+    const testWallets = walletsData.wallets.filter(w => !w.isDeployWallet && w.name !== "deploy-wallet");
+    
+    return { deployWallet: deployWallet || null, testWallets };
+  } catch (error) {
+    console.error("❌ Error loading wallets:", error);
+    return { deployWallet: null, testWallets: [] };
+  }
+}
+
+export function extractAllWallets(): boolean {
+  try {
+    const { deployWallet, testWallets } = loadWallets();
+    const allWallets = deployWallet ? [deployWallet, ...testWallets] : testWallets;
+    
+    if (allWallets.length === 0) {
+      console.error("❌ No wallets found in test-wallets.json");
+      return false;
+    }
+    
+    console.log(`🔧 Extracting ${allWallets.length} wallets...`);
+    
+    let success = true;
+    for (const wallet of allWallets) {
+      try {
+        const keypairArray = wallet.secretKey.array;
+        fs.writeFileSync(wallet.keypairFile, JSON.stringify(keypairArray, null, 2));
+        console.log(`✅ ${wallet.name} → ${wallet.keypairFile}`);
+      } catch (error) {
+        console.error(`❌ Failed to extract ${wallet.name}:`, error);
+        success = false;
+      }
+    }
+    
+    return success;
+  } catch (error) {
+    console.error("❌ Error extracting wallets:", error);
+    return false;
+  }
+}
+
+export function listWallets(): void {
+  try {
+    const { deployWallet, testWallets } = loadWallets();
+    const allWallets = deployWallet ? [deployWallet, ...testWallets] : testWallets;
+    
+    if (allWallets.length === 0) {
+      console.error("❌ No wallets found in test-wallets.json");
+      return;
+    }
+    
+    console.log(`📋 Available wallets (${allWallets.length}):`);
+    allWallets.forEach((wallet, i) => {
+      const status = wallet.isDeployWallet || wallet.name === "deploy-wallet" ? "🔑 Deploy" : "🧪 Test";
+      console.log(`  ${i + 1}. ${status} - ${wallet.name}`);
```

Comment:
i'm getting undefined here

```
📋 Available wallets (3) - Gill version:
  1. 🧪 Test - test-wallet-1
     Public Key: undefined
     Keypair File: test-wallet-1.json
     Balance: Unable to fetch (via Gill)

  2. 🧪 Test - test-wallet-2
     Public Key: undefined
     Keypair File: test-wallet-2.json
     Balance: Unable to fetch (via Gill)

  3. 🧪 Test - test-wallet-3
     Public Key: undefined
     Keypair File: test-wallet-3.json
     Balance: Unable to fetch (via Gill)
     ```
     
     there was an error airdropping stuff so that could be it